aten/src/ATen/detail/HIPHooksInterface.h

```diff
@@ -39,6 +39,14 @@ struct TORCH_API HIPHooksInterface {
     return false;
   }
 
+  virtual bool hasHIPSOLVER() const {
```

Comment:
We shouldn't need this patch if we follow the suggestion in the comment for Context.h